src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -339,7 +353,7 @@ impl<NodeContinue, NodeApp, InnErr, OutErr, ErrHandler>
     TreeVisitorStack<NodeContinue, NodeApp, ErrHandler>
 where
     NodeContinue: Fn(&NodeAppArgs<'_>) -> ContinueTraversal,
-    NodeApp: Fn(NodeAppArgs<'_>) -> Result<(), InnErr>,
+    NodeApp: Fn(UniIndex, AccessRelatedness, &mut TreeVisitor<'_>) -> Result<(), InnErr>,
```

Comment:
So, uh, this entirely removes the purpose of `NodeAppArgs`? Why not extend that type instead?